proposed/http-factory/http-factory.md

```diff
@@ -77,24 +82,30 @@ Has the ability to create server requests.
 ```php
 namespace Psr\Http\Message;
 
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+
 interface ServerRequestFactoryInterface
 {
     /**
      * Create a new server request.
      *
-     * @param string $method
-     * @param UriInterface|string $uri
+     * The method and URI of the request SHOULD be derived from the given
+     * server parameters.
      *
-     * @return ServerRequestInterface
-     */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from PHP globals.
+     * If the method or URI parameter is passed, these values MUST be used
```

Comment:
Actually I'm not sure about rationale of providing the same parameter twice - once in `$server` and then overwriting it in `$method` and `$uri`. Seems like a bad API design to me.